docs/book/src/architecture/controllers/cluster.md

```diff
@@ -1 +1,45 @@
-# Cluster
+# Cluster Controller
+
+The cluster controller is responsible for keeping the cluster object in sync with the infrastructure cluster object.
+
+Its main responsibilities are:
+
+* Setting an OwnerReference on the infrastructure object.
+* Deleting all related objects in the correct order.
```

Comment:
Should we call out that we're effectively implementing foreground deletion?

Reply:
Could do, not sure it's super necessary for core. I'd definitely want it called out on infra providers though.

I'll add some more detail and you can see how it looks and make a call then